File: practice/projects/expense-app-manager-aws/src/services/spend_service.py
import decimal

from models import base_repository, spend


class SpendService:
    def __init__(self, repository: base_repository.BaseRepositorySpend):
        self.repository = repository

    def create_spend(
        self, name: str, description: str, amount: decimal.Decimal
    ) -> spend.Spend:
        spend_item = spend.Spend.new_spend(
            name=name, description=description, amount=amount
        )
        self.repository.create_spend(spend_item.as_dict())
        return spend_item  #

    def get_spends(self) -> list[spend.Spend]:
        return self.repository.get_spends()  #

File: practice/projects/expense-app-manager-java/src/main/java/com/alvaro8317/expense/service/SpendService.java
package com.alvaro8317.expense.service;

import com.alvaro8317.expense.model.Spend;
import com.alvaro8317.expense.repository.BaseRepositorySpend;

import java.math.BigDecimal;

public class SpendService {
    private final BaseRepositorySpend repo;

    public SpendService(BaseRepositorySpend repo) {
        this.repo = repo;
    }

    public Spend createSpend(String name, String description, BigDecimal amount) {
        Spend spend = new Spend(name, description, amount);
        repo.createSpend(spend);
        return spend;
    }

    public Spend getSpends(String name, String description, BigDecimal amount) {
        Spend spend = new Spend(name, description, amount);
        repo.getSpends();
        return spend;
    }
}
